server/src/main/java/io/crate/types/BooleanType.java

```diff
@@ -79,17 +79,23 @@ public Query rangeQuery(String field,
                                 boolean includeUpper,
                                 boolean hasDocValues,
                                 boolean isIndexed) {
-            upperTerm = upperTerm == null || upperTerm;
-            lowerTerm = lowerTerm != null && lowerTerm;
             if (isIndexed) {
                 return new TermRangeQuery(
                     field, indexedValue(lowerTerm), indexedValue(upperTerm), includeLower, includeUpper);
             } else {
                 assert hasDocValues == true : "hasDocValues must be true for Boolean types since 'columnstore=false' is not supported.";
-                return SortedNumericDocValuesField.newSlowRangeQuery(
-                    field,
-                    lowerTerm ? 1 : 0,
-                    upperTerm ? 1 : 0);
+
```

Comment:
Can we detect up-front if the lower or upper terms are out of the range [1, 2] and do a fast conversion to a FieldExistsQuery or a MatchNoDocsQuery?  For example, `a > true` shouldn't match anything, whereas `a <= true` will match everything that has a value for the field.

Reply:
Thank you @romseygeek . was actually wondering if a query like `[2 TO 9223372036854775807]` would be less performant. I will try to use FieldExistsQuery or a MatchNoDocsQuery.